src/functTest/groovy/com/bmuschko/gradle/docker/tasks/container/DockerCreateContainerFunctionalTest.groovy

```diff
@@ -167,6 +167,48 @@ class DockerCreateContainerFunctionalTest extends AbstractGroovyDslFunctionalTes
         build('inspectStoppedContainer')
     }
 
+    def "with publishAll set, all ports get a host binding"() {
+        given:
+        def exposedPorts = [1234, 2345]
+        String containerCreationTask = """
+            task createContainer(type: DockerCreateContainer) {
+                dependsOn pullImage
+                targetImageId pullImage.getImageId()
+                autoRemove = true
+                
+                cmd = ['sleep', '10']
+                exposePorts 'tcp', $exposedPorts
+                publishAll = true
+            }
+        """
+
+        String containerInspect = '''
+            task inspectContainer(type: DockerInspectContainer) {
+                dependsOn startContainer
+                targetContainerId startContainer.getContainerId()
+                
+                onNext { container ->
+                  container.networkSettings.ports.bindings.forEach { exposedPort, bindings ->
+                    logger.quiet "$exposedPort.port -> ${bindings.first().hostPortSpec}"
+                  }
+                }
+                finalizedBy stopContainer
+            }
+        '''
+
+        buildFile << containerStart(containerCreationTask)
+        buildFile << containerStop()
+        buildFile << containerInspect
+
+        when:
+        BuildResult buildResult = build('inspectContainer')
+
+        then:
+        exposedPorts.every {
+            buildResult.output.find(/$it -> \d+/) != null
```

Comment:
open to a cleaner solution for capturing the output of docker inspect; not familiar with gradle's testing APIs and saw this kind of thing done elsewhere.

Reply:
Could we verify the actual ports that we'd expect?

Reply:
how do you suggest I do that?

Reply:
Oh, I see. Then it's probably not possible. I think we should be OK with merging what we have right now.

@cdancy Do you have any other ideas here?